Modules/Config/ConfigModule.lua

```diff
@@ -98,18 +98,59 @@ local options = {
                 MinimapModule:UpdateMinimap()
             end
         },
-        mapPinsHeader = {
+        brokerDescription = {
             order = 8,
+            name = "LibDataBroker settings" .. "\n",
+            type = "description"
+        },
+        brokerGoal = {
+            order = 9,
+            name = "Display Goal",
+            type = "select",
+            width = standardControlWidth,
+            desc = "The goal to track in the data source.",
+            values = {
+                G1COLLECT = "Collect at least one",
+                G2COLLECTRARE = "Collect at least one rare",
+                G3COLLECTMAX = "Collect maximum amount",
+                G4COLLECTMAXRARE = "Collect maximum amount rare",
```

Comment:
@GurliGebis Is there a reason the addon uses prefixes like "T1" (or in this case "G1") for drop-down config values?  My assumption was that it was to set the ordering of the options.  If so, we could consider dropping the prefix and instead use the "sorting" option we're now using for mapPinsToInclude and forfeitPromptUnless.  Of course, changing that would require a config migration or breaking config compatibility (which I hate).

Reply:
It is for sorting - I didn't know about the sorting option (or I missed it when I read the documentation) when I implemented it.
But can we change it, without people losing thier current setting?

Reply:
I think so. I'll look into it.

Reply:
Thinking about it - it might be possible to change the "get" function to handle it, so if it sees the old value, it will just replace it with the new value.
However, that would mean we would have to carry the old values there forever - not sure that is the best solution (though I cannot think of anything better at the moment)

Reply:
That would work.  The other main option I'm aware of is to centralize handling of "old config" and migrate to the current version on addon initialization.

Here's a few examples of addons who use that approach (some with more sophistication than others):

* https://github.com/RealUI/RealUI/blob/11e630e04986c7947b0b33e6f4b66b9b1ad4e238/RealUI_Inventory/RealUI_Inventory.lua#L76
* https://github.com/Urtgard/WQAchievements/blob/master/WQAchievements.lua#L408
* https://github.com/TorelTwiddler/CanIMogIt/blob/6a6021d8572b607e73f9aa7fb8c64bb9b70f74b1/database.lua#L130

Reply:
The reason I'd prefer centralizing is that it makes it so that the legacy logic is easy to find, and can easily be removed when it's either too old, or something happens (such as a new plugin for a different version of the game, or changing plugin ownership) where legacy config support is no longer needed.

Reply:
That makes sense - can I get you to add it to this branch, and then change all the places where the order is included using the name?

Reply:
If you're okay with it, how about I clean up what I have in this branch so far, we work on getting it merged, and then I work on separate PRs for adding config migration and the separate goal tracking window?

It's getting big enough that it will be tricky to review.

Reply:
That is indeed a better solution 👍